jenkins/agent-ctl.sh

```diff
@@ -0,0 +1,248 @@
+#!/bin/bash
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Run command with no args for usage instructions.
+
+usage() {
+  echo "Usage: $(basename "${0}") [--large] <instance> [AUTO|create|delete|update|reboot|attach|detatch]"
+  exit 1
+}
+
+
+set -o nounset
+set -o errexit
+
+DOCKER_VERSION='1.9.1-0~wheezy'
+GO_VERSION='go1.6.2.linux-amd64.tar.gz'
+TIMEZONE='America/Los_Angeles'
+
+# Defaults
+DISK_SIZE='500GB'  # TODO(fejta): 300G for local-ssd
+DISK_TYPE='pd-ssd'   # TODO(fejta): local-ssd
+IMAGE='debian-7-backports'  # TODO(fejta): debian8
+LARGE=
+MACHINE_TYPE='n1-highmem-32'  # TODO(fejta): optimize to lower values
+MASTER='jenkins-master'
+SCOPES='cloud-platform,compute-rw,storage-full'  # TODO(fejta): verify
+
+if [[ -z "${1:-}" ]]; then
+  usage
+fi
+
+while true; do
+  case "${1:-}" in
+    --large)
+      LARGE=yes
+      shift
+      ;;
+    --pr)
+      echo 'Talking to PR jenkins'
+      MASTER='pull-jenkins-master'
+      shift
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+if [[ -z "${LARGE}" ]]; then
+  read -p "Using ${MACHINE_TYPE} for testing. Continue [Y/n]: " ans
+  if [[ ! "${ans}" =~ '^[yY]' ]]; then
+    echo 'Add --large'
+    exit 1
+  fi
+  DISK_SIZE='200GB'
+  DISK_TYPE='pd-standard'
+  MACHINE_TYPE='n1-standard-1'
```

Comment:
I don't understand this. If I don't pass `--large`, it prints out
`Using n1-highmem-32 for testing. Continue [Y/n]: `

If I type `y`, it then changes to an `n1-standard-1`??